js/handlers.js

```diff
@@ -20,12 +20,27 @@ weechat.factory('handlers', ['$rootScope', '$log', 'models', 'plugins', 'notific
     };
 
     // inject a fake buffer line for date change
-    var injectDateChangeMessage = function(message, buffer, date) {
-        var content = "Date changed to " + date.toDateString();
+    var injectDateChangeMessage = function(message, buffer, d1, d2) {
+        var d1_plus_one = d1;
+        d1_plus_one.setDate(d1_plus_one.getDate() + 1);
+        d1.setHours(0, 0, 0, 0);
+        d2.setHours(0, 0, 0, 0);
+        // it's not always true that a date with time 00:00:00
+        // plus one day will be time 00:00:00
+        d1_plus_one.setHours(0, 0, 0, 0);
+        var content = "";
+        if (d1_plus_one.valueOf() === d2.valueOf()) {
+            content = "\u001943Date changed to " + d2.toDateString();
```

Comment:
`\u001943` corresponds to `chat_day_change` in `weechat.js`'s `WeeChatProtocol._colorsOptionsNames`, want me to make a PR for that comment?